project/frontend/src/pages/SwipeMatchPage.js

```diff
@@ -0,0 +1,185 @@
+import React, { useState, useEffect, useContext, useReducer } from 'react';
+
+import { AuthContext } from '../components/Authentication.js';
+import FirestoreContext from '../contexts/FirestoreContext.js';
+import PlacesApiContext from '../contexts/PlacesApiContext.js';
+
+import Modal from '../components/Modal.js';
+import RestaurantCardDeck from '../components/RestaurantCardDeck.js';
+import SwipeMatchConsole from '../components/SwipeMatchConsole.js';
+import SwipeMatchLeaderboard from '../components/SwipeMatchLeaderboard.js';
+
+import SwipeMatchService from '../scripts/SwipeMatchFirestoreService.js';
+
+function reducer(previous, action) {
+  let { location, restaurants, sessionStarted, users } = previous;
+  const { payload, type } = action;
+
+  if (type === 'set restaurants') {
+    restaurants = payload;
+  } else if (type === 'snapshot') {
+    if (
+      location.lat !== payload.location.lat ||
+      location.lng !== payload.location.lng
+    ) {
+      location = payload.location;
+    }
+
+    if (sessionStarted !== payload.sessionStarted) {
+      sessionStarted = payload.sessionStarted;
+    }
+
+    if (users.toString() !== payload.users.toString()) {
+      users = payload.users;
+    }
+
+    restaurants = restaurants.map((restaurant) => {
+      const likes = payload.likes[restaurant.key.id];
+      if (likes) {
+        restaurant.fractionLiked = likes.length / users.length;
+      }
+      return restaurant;
+    });
+  }
+
+  return { location, restaurants, sessionStarted, users };
+}
+
+function SwipeMatchPage(props) {
+  const { currLocation, action, username } = props.location.state;
+  const isCreator = action === 'create';
+
+  const authContext = useContext(AuthContext);
+  const { firebase, firestore } = useContext(FirestoreContext);
+  const placesService = useContext(PlacesApiContext);
+
+  const [groupId, setGroupId] = useState(props.location.state.groupId || '');
+  const [
+    { location, restaurants, sessionStarted, users },
+    dispatch,
+  ] = useReducer(reducer, {
+    location: { lat: undefined, lng: undefined },
+    restaurants: [],
+    sessionStarted: false,
+    users: [],
+  });
+
+  useEffect(
+    () => {
+      let unsubscribe;
+      let id = groupId;
+
+      (async () => {
+        if (isCreator) {
+          let user = authContext.currentUser.get;
+
+          if (user && user.isSignedIn()) {
+            let creatorCurrentSwipeMatchSession = null;
+            try {
+              creatorCurrentSwipeMatchSession = await SwipeMatchService.fetchCreatorCurrentSwipeMatchSession(
+                firestore,
+                user
+              );
+            } catch (e) {}
```

Comment:
if its intentional to do nothing on catching the error here, then lets leave a comment with // do nothing, otherwise a TODO on how we'd handle it

Reply:
Ended up refactoring this whole section out, so resolved!